webhooks-extension/src/components/WebhookDisplayTable/tests/WebhookDeleteNotifications.test.js

```diff
@@ -0,0 +1,129 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { waitForElement, cleanup, fireEvent } from 'react-testing-library';
+import { WebhookDisplayTable } from '../WebhookDisplayTable'
+import * as API from '../../../api';
+import { renderWithRouter } from '../../../test/utils/test'
+import 'react-testing-library/cleanup-after-each';
+
+beforeEach(() => {
+  jest.restoreAllMocks
+  jest.resetModules()
+});
+
+afterEach(() => {
+  jest.clearAllMocks()
+  cleanup()
+});
+
+function fakeDeleteWebhooksSuccess() {
+  return Promise.resolve({
+    data: {},
+    status: 204,
+    statusText: 'OK',
+    headers: {}
+  });
+}
+
+function fakeDeleteWebhooksFailure() {
+  return Promise.reject({
+    data: {},
+    status: 400,
+    statusText: 'Error',
+    headers: {}
+  });
+}
+
+const fakeRowSelection = [
+  {
+    "id":"mywebhook|default",
+    "isSelected":true,
+    "isExpanded":false,
+    "cells":[
+      {
+        "id":"mywebhook|default:name","value":"mywebhook","isEditable":false,"isEditing":false,"isValid":true,"errors":null,"info":
+        {
+          "header":"name"
+        }
+      },
+      {
+        "id":"mywebhook|default:repository","value":"https://github.com/foo/bar","isEditable":false,"isEditing":false,"isValid":true,"errors":null,"info":
+        {
+          "header":"repository"
+        }
+      },
+      {
+        "id":"mywebhook|default:pipeline","value":"simple-helm-pipeline-insecure","isEditable":false,"isEditing":false,"isValid":true,"errors":null,"info":
+        {
+          "header":"pipeline"
+        }
+      },
+      {"id":"mywebhook|default:namespace","value":"default","isEditable":false,"isEditing":false,"isValid":true,"errors":null,"info":
+      {
+        "header":"namespace"
+      }
+    }
+  ]}
+]
+
+const webhooks = [
+  {
+    id: '0|namespace',
+    name: 'first test webhook',
+    gitrepositoryurl: 'repo1',
+    pipeline: 'pipeline1',
+    namespace: 'namespace1'
+  }
+];
+
+it('should display a success message on a good delete', async () => {
+  let getWebhooksMock = jest.spyOn(API, "getWebhooks").mockImplementation(() => Promise.resolve(webhooks));
+  let getRowsMock = jest.spyOn(API, "getSelectedRows").mockImplementation(() => fakeRowSelection);
+  let deleteWebhooksMock = jest.spyOn(API, "deleteWebhooks").mockImplementation(() => fakeDeleteWebhooksSuccess);
+  const { getByText } = renderWithRouter(<WebhookDisplayTable match={{}} />);  
+  await waitForElement(() => getByText('first test webhook'));  
+  const deleteButton = getByText('Delete');
+  fireEvent.click(deleteButton);
+  const deleteWebhookButton = getByText('Delete webhook');
+  fireEvent.click(deleteWebhookButton);
+  expect(getWebhooksMock).toHaveBeenCalled();
+  expect(getRowsMock).toHaveBeenCalled();
+  expect(deleteWebhooksMock).toHaveBeenCalled();
+  await waitForElement(() => getByText('Webhook(s) deleted successfully.'));
```

Comment:
Might be worth checking things that are supposed to have appeared ... were not previously there (probably applies to multiple tests)

Reply:
Agreed, added